src/OmniSharp/Workers/Refactoring/FixUsings.cs

```diff
@@ -0,0 +1,212 @@
+                    {
+                        if (codeOperation != null)
+                        {
+                            codeOperation.Apply(_workspace, CancellationToken.None);
+                        }
+                    }
+                }
+            }
+
+            return;
+        }
+
+        private async Task SortUsings()
+        {
+            //Sort usings
+            var nRefactoryProvider = new NRefactoryCodeActionProvider();
+            var sortActions = new List<CodeAction>();
+            var refactoringContext = await GetRefactoringContext(_document, sortActions);
+            if (refactoringContext != null)
+            {
+                var sortUsingsAction = nRefactoryProvider.Refactorings
+                    .First(r => r is ICSharpCode.NRefactory6.CSharp.Refactoring.SortUsingsAction);
+
+                await sortUsingsAction.ComputeRefactoringsAsync(refactoringContext.Value);
+
+                foreach (var action in sortActions)
+                {
+                    var operations = await action.GetOperationsAsync(CancellationToken.None).ConfigureAwait(false);
+                    if (operations != null)
+                    {
+                        foreach (var codeOperation in operations)
+                        {
+                            if (codeOperation != null)
+                            {
+                                codeOperation.Apply(_workspace, CancellationToken.None);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        private static async Task<CodeRefactoringContext?> GetRefactoringContext(Document document, List<CodeAction> actionsDestination)
+        {
+            var firstUsing = (await document.GetSyntaxTreeAsync()).GetRoot().DescendantNodes().FirstOrDefault(n => n is UsingDirectiveSyntax);
+            if (firstUsing == null)
+            {
+                return null;
+            }
+            var location = firstUsing.GetLocation().SourceSpan;
+
+            return new CodeRefactoringContext(document, location, (a) => actionsDestination.Add(a), CancellationToken.None);
+        }
+
+        private async Task<IEnumerable<CodeActionOperation>> GetUsingActions(ICodeActionProvider codeActionProvider,
+                ImmutableArray<Diagnostic> pointDiagnostics, string actionPrefix)
+        {
+            var actions = new List<CodeAction>();
+            var context = new CodeFixContext(_document, pointDiagnostics.First().Location.SourceSpan, pointDiagnostics, (a, d) => actions.Add(a), CancellationToken.None);
+            var providers = codeActionProvider.CodeFixes;
+
+            //Disable await warning since we dont need the result of the call. Else we need to use a throwaway variable.
+#pragma warning disable 4014
+            foreach (var provider in providers)
+            {
+                provider.RegisterCodeFixesAsync(context);
```

Comment:
@nosami, That is because the async method returns a Task which we need to ignore. But ignoring the returned object causes the compiler to throw a warning. So another option is to do an await and read the return value into a throwaway variable. I had that in place before, but used the pragma to be explicit. http://stackoverflow.com/a/12548725